src/codeu/chat/server/Controller.java

```diff
@@ -152,6 +174,26 @@ public User login(Uuid id, String username, String password, Time creationTime)
           result);
     }
 
+    //Load in all of the user's conversations and messages
+    ArrayList<ConversationData> conversations = new ArrayList<ConversationData>();
+    conversations = storage.loadConversations(username);
+    for (ConversationData c: conversations){
+      Uuid convoId = createId();
+      Conversation currentConvo = newConversation(convoId, c.getTitle(), id, c.getCreation());
+      conversationIds.put(convoId, c.getId());
```

Comment:
From a readability standpoint, this code block is little confusing. If you're doing a null check on `currentConvo`, do it as soon as you have assigned value to it (i.e the result of `newConversation()`).

If you need to add the ID to `conversationIds`, do that separately, either in the beginning of the for loop or after the if-block.

It is a good practice to define and assign variables closest to the point you will start using them.